crates/cranelift/src/func_environ.rs

```diff
@@ -1839,6 +1839,15 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
                 u64::MAX
             });
 
+        let max_size = self.module.memory_plans[index].memory.maximum.map(|max| {
+            max.checked_mul(u64::from(WASM_PAGE_SIZE))
+                .unwrap_or_else(|| {
+                    // See comment in `min_size` computation above.
+                    debug_assert_eq!(max, 1 << 48);
+                    u64::MAX
+                })
+        });
```

Comment:
Might be better to `and_then` instead of `map` here actually and not swap out the max for `u64::MAX` on overflow?

Reply:
Done.